### PR TITLE
feat: to navigate through videos by prev and next video button

### LINKED ISSUE #1166: feature: Add "Next" and "Previous" Buttons for Module Navigation
image

**Description**:
We need to implement "Next" and "Previous" buttons below the video to allow users to easily navigate between modules and courses.

**Scope**:
1. Add two buttons, "Next" and "Previous," below the video in the course module page.
2. The buttons should allow users to:
   - Move to the next module if available.
   - If there are no more modules in the current course, navigate to the next course.
   - Move to the previous module if available.
   - If at the first module, navigate to the previous course’s last module.
   - If no further modules or courses are available, disable the respective button.

**Acceptance Criteria**:
1. "Next" button should navigate to the next module in the current course.
2. If the last module of the course is reached, the "Next" button should navigate to the first module of the next course.
3. "Previous" button should navigate to the previous module in the current course.
4. If at the first module of the course, the "Previous" button should navigate to the last module of the previous course.
5. Both buttons should be disabled when there are no further modules or courses.
6. Buttons should be visible below the video player on the module page.

**UI/UX Considerations**:
- Place buttons directly below the video player.
- Ensure the button design is consistent with the overall UI theme.
- Add hover and disabled states to indicate functionality.

### PR DESCRIPTION
**Pull request by Prashant-flick**
### PR Features:
- Added functionality to navigate through videos within a course using previous and next buttons.
- Automatically plays the next video in the course after the current video ends.

Limitations
- Navigation to the next course after the last video in the current course is not implemented.
- Navigation is limited to videos only, excluding notes.

https://github.com/user-attachments/assets/47436014-6c33-4714-ae3a-4eef288395d8

Resolves #1166

- Open to any suggestions for improvements; please let me know your thoughts in the comments.

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by devsargam** [comment]
adding it to things to review

**Comment by Prashant-flick** [comment]
hey @devsargam 
your thoughts on this?

**Comment by devsargam** [comment]
This is great but I need to review for all edge cases

**Review by devsargam — CHANGES_REQUESTED** [review]

File: src/components/VideoPlayer2.tsx [hunk 1]
```diff
@@ -64,34 +71,29 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
...
-        event.preventDefault();
-        event.stopPropagation();
-        togglePictureInPicture();
-      },
-    });
+    const pipButtonContainer = (player as any).controlBar.addChild('button');
 
     const root = createRoot(pipButtonContainer.el());
-    root.render(<PipButton />);
+    root.render(<span className="material-symbols-outlined">picture_in_picture_alt</span>);
```
[the remark this answers is not in the document]
Comment (Prashant-flick) [reply src/components/VideoPlayer2.tsx:78]: there was some error on prev logic, and also some error on my logic.
i have fixed all the errors, please review it.

**Comment by devsargam** [comment]
Sorry this looks really good but considering this pr
ref: https://github.com/code100x/cms/pull/1434

**Pull request closed without merging**